envelope.go

```diff
@@ -59,7 +127,9 @@ type envelopeWriter struct {
 
 func (w *envelopeWriter) Marshal(message any) *Error {
 	if message == nil {
-		if _, err := w.writer.Write(nil); err != nil {
+		// Send a zero-length message.
```

Comment:
This is misleading since a zero-length message on a stream is still a message, and this is really sending _nothing_ (since it elides any envelope).
```suggestion
		// Send a no-op (just flushing headers/making sure request sent)
```

Reply:
Reworded as: `// Send no-op message to create the request and send headers.`